src/P2P/Node/PeerDB.hs

```diff
@@ -429,7 +428,7 @@ updatePeerDb (PeerDb _ _ lock var) a f
     = withMVar lock . const . atomically . modifyTVar' var $ \s ->
         case getOne $ getEQ a s of
             Nothing -> s
-            Just x -> let !r = f x in updateIx a r s
+            Just x -> force $ updateIx a (force $ f x) s
```

Comment:
I know we keep saying "use the force", but you could probably just force this once 😂

Reply:
Depends on the inner data structures. I don't recall how NFData for MVar is implemented (or if it is implemented at all). But I remember running into issues with respect to forcing across references, thread bounds, in in particular uses of STM.

I don't recall what was the reason for the force in this particular line. But all the uses of `force` in this file date back to a profiling session with @jwiegley several month ago which resulted in huge speed ups for catching up nodes.